server/handler.go

```diff
@@ -939,54 +939,51 @@ func (h *Handler) IsLeader() bool {
 
 // PackHistoryHotReadRegions get read hot region info in HistoryHotRegion form.
 func (h *Handler) PackHistoryHotReadRegions() (historyHotRegions []core.HistoryHotRegion, err error) {
-	hotReadLeaderRegions := h.GetHotReadRegions().AsLeader
-	historyLeaderHotRegions, err := h.packHotRegions(hotReadLeaderRegions, true, core.HotRegionTypes[0])
-	if err != nil {
+	hotReadRegions := h.GetHotReadRegions()
+	if hotReadRegions == nil {
 		return
 	}
-	hotReadPeerRegions := h.GetHotReadRegions().AsPeer
-	historyPeerHotRegions, err := h.packHotRegions(hotReadPeerRegions, false, core.HotRegionTypes[0])
+	hotReadPeerRegions := hotReadRegions.AsPeer
+	historyPeerHotRegions, err := h.packHotRegions(hotReadPeerRegions, core.ReadType.String())
 	if err != nil {
 		return
 	}
-	historyHotRegions = append(historyHotRegions, historyLeaderHotRegions...)
 	historyHotRegions = append(historyHotRegions, historyPeerHotRegions...)
 	return
 }
 
 // PackHistoryHotWriteRegions get write hot region info in HistoryHotRegion from
 func (h *Handler) PackHistoryHotWriteRegions() (historyHotRegions []core.HistoryHotRegion, err error) {
-	hotWriteLeaderRegions := h.GetHotWriteRegions().AsLeader
-	historyLeaderHotRegions, err := h.packHotRegions(hotWriteLeaderRegions, true, core.HotRegionTypes[1])
-	if err != nil {
+	hotWriteRegions := h.GetHotWriteRegions()
+	if hotWriteRegions == nil {
 		return
 	}
-	hotWritePeerRegions := h.GetHotWriteRegions().AsPeer
-	historyPeerHotRegions, err := h.packHotRegions(hotWritePeerRegions, false, core.HotRegionTypes[1])
+	hotWritePeerRegions := hotWriteRegions.AsPeer
+	historyPeerHotRegions, err := h.packHotRegions(hotWritePeerRegions, core.WriteType.String())
 	if err != nil {
 		return
 	}
-	historyHotRegions = append(historyHotRegions, historyLeaderHotRegions...)
 	historyHotRegions = append(historyHotRegions, historyPeerHotRegions...)
 	return
 }
 
-func (h *Handler) packHotRegions(hotPeersStat statistics.StoreHotPeersStat, isLeader bool, hotRegionType string) (historyHotRegions []core.HistoryHotRegion, err error) {
+func (h *Handler) packHotRegions(hotPeersStat statistics.StoreHotPeersStat, hotRegionType string) (historyHotRegions []core.HistoryHotRegion, err error) {
 	c, err := h.GetRaftCluster()
 	if err != nil {
 		return nil, err
 	}
 	for _, hotPeersStat := range hotPeersStat {
 		stats := hotPeersStat.Stats
 		for _, hotPeerStat := range stats {
-			region := c.GetRegion(hotPeerStat.RegionID).GetMeta()
-			region, err := encryption.EncryptRegion(region, h.s.encryptionKeyManager)
+			region := c.GetRegion(hotPeerStat.RegionID)
+			meta := region.GetMeta()
```

Comment:
region may be nil

Reply:
solved

Reply:
it may return nil after c.GetRegion(hotPeerStat.RegionID), so meta := region.GetMeta() will panic

Reply:
sorry,I forget to update it